index.js

```diff
@@ -6,57 +6,125 @@ var types = [
   'string',
   'symbol',
   'boolean',
+  'date',
   'function', // Weird to expose this
   'undefined',
 ];
 
+// Add methods for each type
+types.forEach(function(type) {
+  normalize[type] = normalize.bind(null, type);
+});
+
+
 function normalize(type, value) {
-  var args = Array.prototype.slice.call(arguments, 2);
+  if (typeof value === 'function') {
+    if (type === 'function') {
+      return value;
+    }
+    value = value.apply(null, Array.prototype.slice.call(arguments, 2));
+  }
+  return coerce(type, value);
+}
 
-  var isType = conforms(type, value);
-  var isFunction = typeof value === 'function';
 
-  if (!isType && !isFunction) {
+function primitive(value) {
+  if (value !== null && typeof value === 'object') {
```

Comment:
Ah! overlooked that possibility. I will also need a guard against `.toString` being absent.

I'll have some time to fix things up tomorrow.